src/c++/perf_analyzer/report_writer.cc

```diff
@@ -209,7 +217,16 @@ ReportWriter::GenerateReport()
           ofs << avg_latency_us << ",";
         }
         ofs << std::to_string(avg_send_time_us + avg_receive_time_us) << ",";
-        ofs << std::to_string(avg_response_wait_time_us);
+        ofs << std::to_string(avg_response_wait_time_us) << ",";
+        if (should_output_metrics_) {
+          if (status.metrics.size() == 1) {
+            WriteGpuMetrics(ofs, status.metrics[0]);
+          } else {
+            std::cerr << "There should only be one entry in the metrics vector."
```

Comment:
How about making the error message part of the exception object?

Reply:
Putting off for future refactor of `PerfAnalyzerException`. It should include a message.